lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -0,0 +1,133 @@
+from http import HTTPStatus
+import os
+import werkzeug.utils
+import typing as t
+from datetime import datetime
+from pathlib import Path
+
+import pytest
+import requests
+
+if t.TYPE_CHECKING:
```

Comment:
Is there a reason for conditionalizing these types?

And as I saw this, I also realize that I hadn't even consciously noticed you were importing all of `typing` and using `t.Union` and such throughout. That wastes a bit more space, and it's not the way we do type hints in every other file in the repo, which makes it look a bit odd... and potentially makes the file just a bit harder to read.

I'd prefer you do this like everywhere else, and just `from typing import Callable, Union, ...`